src/main/java/com/wynprice/secretrooms/client/compat/SecretRoomsWailaOverride.java

```diff
@@ -0,0 +1,29 @@
+package com.wynprice.secretrooms.client.compat;
+
+import com.wynprice.secretrooms.server.blocks.SecretBaseBlock;
+import com.wynprice.secretrooms.server.items.TrueVisionGogglesClientHandler;
+import mcp.mobius.waila.api.IBlockAccessor;
+import mcp.mobius.waila.api.IBlockComponentProvider;
+import mcp.mobius.waila.api.IPluginConfig;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.Blocks;
+import net.minecraft.item.ItemStack;
+
+public class SecretRoomsWailaOverride implements IBlockComponentProvider {
+    @Override
+    public BlockState getOverride(IBlockAccessor accessor, IPluginConfig config) {
+        if(TrueVisionGogglesClientHandler.isWearingGoggles(accessor.getPlayer()))
+            return accessor.getBlockState();
+        return SecretBaseBlock.getMirrorState(accessor.getWorld(), accessor.getPosition()).orElse(Blocks.STONE.getDefaultState());
```

Comment:
Instead of passing STONE to the `orElse`, could you pass `accessor.getBlockState()`

Reply:
Sure.